### PR TITLE
FIXED#1377  App returns from Settings to main activity on back pressed

### LINKED ISSUE #1377: App should return back to the Main Activity from the Device Settings activity
**Describe the feature you'd like**
App should return back to the Main Activity from the Device Settings activity on back pressed presently it goes back to the device's home screen.

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by codepoet2017390**
Fixes #1377
Created a back to main function in Settings Fragment and handled the pass of control in the Main Activity on Back Pressed.
Checklist:

- [ ] I have checked for PMD and check-style issues 
- [ -] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [- ] My branch is up-to-date with the Upstream development branch.

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java [hunk 1]
```diff
@@ -111,9 +112,18 @@ public void onBackPressed() {
...
+            if(f instanceof SettingsFragment)
+            {
+                finish();
+                ((SettingsFragment) f).goToMain();
+            }
+            else {
+                finish();
+            }
+        }
+        else{
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/main/MainActivity.java]: This is not a fix. This is a workaround

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 2]
```diff
@@ -101,4 +102,10 @@ public void onResume() {
         ViewUtils.setTitle(this, getString(R.string.device_settings));
         setSalesDataSummary();
     }
+
+    public void goToMain()
+    {
+        Intent intent = new Intent(getActivity(), MainActivity.class);
+        startActivity(intent);
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: Not according to Java style guidelines, this is not C++

File: app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java [hunk 3]
```diff
@@ -101,4 +102,9 @@ public void onResume() {
         ViewUtils.setTitle(this, getString(R.string.device_settings));
         setSalesDataSummary();
     }
+
+    public void BackToMainActivity() {
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: This is not C#

[the remark this answers is not in the document]
Comment (codepoet2017390) [reply app/src/main/java/com/eventyay/organizer/core/settings/SettingsFragment.java]: @iamareebjamal can you please explain clearly what is the issue with the style and what exact style type is needed

**Pull request closed without merging**